datadog_checks_dev/datadog_checks/dev/utils.py

```diff
@@ -82,10 +77,6 @@ def dir_exists(d):
     return os.path.isdir(d)
 
 
-def path_exists(p):
```

Comment:
Why remove this?

Reply:
It's not used anywhere, it adds a useless indirection level, basically it's 2 lines of untested code

Reply:
I had planned on making these path existence functions faster, as that was the main goal. Can we please keep this? I can even do this in a PR right away if you want.

Reply:
Again, I don't really like the indirection level. How faster we can go? Is it really that important to require more code, at the price of readability? It's a dev tool, we don't have hot paths to optimize here...

Reply:
Are we talking about microseconds differences here? These methods are really small. I am no Python expert, I know in the Java world it's not a best practice to do such things (unless if you are using a specific design pattern like the proxy or visitor DP). It adds a layer of abstraction between the real library and the code that uses it. In the end, this need to be documented (it's not right now)

Reply:
Milliseconds since our loops go through each check in repo, sometimes multiple times for a command.

You both might find this interesting https://mail.python.org/pipermail/python-dev/2018-May/153296.html

Reply:
I do indeed:
>Speaking as a Mercurial and Firefox build system developer,
*milliseconds absolutely matter*. Going further, *fractions of
milliseconds matter*. For Mercurial's test suite with its ~25,000 Python
process invocations, 1ms translates to ~25s of CPU time

Now you tell me how close `ddev` is to that use case at this very moment. I won't argue further and we can keep the (untested) indirection level if you like so but this is just plain premature optimization.

> Also are we fine with breaking contributors who happen to be using this to test locally?

Do we have an idea of how many contributors are we talking about? The package is still at version `0.x` so I don't feel like this would be much of a problem.

Reply:
I don't know but https://pepy.tech/project/datadog-checks-dev

The only ci would be -extras (which doesn't get run often) since -core uses master

Reply:
Commit reverted, I'll see if I can document the reason behind the indirection level somewhere in the code